client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions-i2/pending-actions/index.tsx

```diff
@@ -0,0 +1,66 @@
+import { Spinner } from '@wordpress/components';
+import classNames from 'classnames';
+import { translate } from 'i18n-calypso';
+import { FC } from 'react';
+import './style.scss';
+
+interface VisualStateIndicatorProps {
+	state: 'pending' | 'success' | 'error';
+	text: string;
+}
+
+const VisualStateIndicator = ( { state, text }: VisualStateIndicatorProps ) => {
+	let icon: string | JSX.Element;
+	switch ( state ) {
+		case 'pending':
+			icon = <Spinner />;
+			break;
+		case 'success':
+			icon = '✅';
+			break;
+		case 'error':
+			icon = '❌';
+			break;
+		default:
+			icon = '';
+			break;
+	}
+	return (
+		<>
+			{ state === 'pending' && <i>{ text }</i> }
+			{ state !== 'pending' && text }
+			<span className="visual-state-indicator">{ icon }</span>
+		</>
+	);
+};
+
+interface Props {
```

Comment:
Optional:
What do you think about use an array of tasks, progress and title, instead of to define one by one?

Reply:
That can be a good follow up. For the time being there are no more actions coming so it's not really necessary right now.